packages/flutter/lib/src/rendering/list_wheel_viewport.dart

```diff
@@ -562,34 +626,109 @@ class RenderListWheelViewport
+        // Pre-transform painting function.
+            (PaintingContext context, Offset offset) {
+          context.paintChild(
+            child,
+            offset + new Offset(
+              untransformedPaintingCoordinates.dx,
+              // Paint everything in the center (e.g. angle = 0), then transform.
+              -_topScrollMarginExtent,
+            ),
+          );
+        },
+      );
+    }
+
+    if (!useMagnifier) {
+      _paintCylindrically(context, offset);
+      return;
+    }
+
+    final double magnifierTopLinePosition =
+        size.height / 2 - _itemExtent * _magnifyRate / 2;
+    final double magnifierBottomLinePosition =
+        size.height / 2 + _itemExtent * _magnifyRate / 2;
+
+    // Some part of the child is in the center magnifier.
+    if (untransformedPaintingCoordinates.dy
+          >= magnifierTopLinePosition - _itemExtent * _magnifyRate
+        && untransformedPaintingCoordinates.dy <= magnifierBottomLinePosition) {
+      final Rect centerRect = new Rect.fromLTWH(
+          0.0,
+          magnifierTopLinePosition,
+          size.width,
+          _itemExtent * _magnifyRate);
+      final Rect topHalfRect = new Rect.fromLTWH(
+          0.0,
+          0.0,
+          size.width,
+          magnifierTopLinePosition);
+      final Rect bottomHalfRect = new Rect.fromLTWH(
+          0.0,
+          magnifierBottomLinePosition,
+          size.width,
+          magnifierTopLinePosition);
+
+      // Clipping the part in the center.
+      context.pushClipRect(
+          false,
+          offset,
+          centerRect,
+          (PaintingContext context, Offset offset) {
+            context.pushTransform(
+                false,
+                offset,
+                _magnifyTransform(),
+                (PaintingContext context, Offset offset) {
+                  context.paintChild(
+                      child,
+                      offset + untransformedPaintingCoordinates);
+                });
+          });
+
+      // Clipping the part in either the top-half or bottom-half of the wheel.
+      context.pushClipRect(
+          false,
+          offset,
+          untransformedPaintingCoordinates.dy <= magnifierTopLinePosition
+              ? topHalfRect
+              : bottomHalfRect,
+          (PaintingContext context, Offset offset) {
+            _paintCylindrically(context, offset);
+          }
+      );
+    }
+    else
+      _paintCylindrically(context, offset);
+  }
+
+  /// Return the Matrix4 transformation that would zoom in the center item.
+  Matrix4 _magnifyTransform() {
+    final Matrix4 magnify = Matrix4.identity();
+    magnify.translate(size.width / 2 * (_curveRate + 1), size.height / 2);
+    magnify.scale(_magnifyRate, _magnifyRate, _magnifyRate);
+    magnify.translate(-size.width / 2 * (_curveRate + 1), -size.height / 2);
+    return magnify;
   }
 
   /// Apply incoming transformation with the transformation's origin at the
   /// viewport's center.
```

Comment:
or horizontally off to the side when offAxisFraction is not `0.0`.

Reply:
fixed.